### ISSUE TITLE
Splitting "proxies/scanners" from BLE "trackers" and every other random BT device

### ISSUE THREAD

**Issue by wappinghigh** [agittins/bermuda#276]
When you go to configure>set devices, I get something like : Bermuda can currently see 31 active of 170 bluetooth devices, reported across 3 active of 4 bluetooth scanner devices. Life looks good.

With my Shelly BT nodes these were easy to find in the drop down list. Therefore easy to add (I have 3 BT capable Shellys) They were labelled as "Shelly xxxx.x.xx etc " in the list 

But I just can't seem to see/find the 4th "scanner device" from all the other reported BT noise...

Is there any way in set up, to separate fixed BT scanner devices from all the other random BLE's that seem to come and go?

Thanks!

**Comment by agittins** [comment]
I can see how that would be a bit confusing. The drop-down list is for selecting "devices" that **transmit** advertisements, so you don't *need* to select your proxies/scanners in that list.

The Shelly devices are a bit different in that they are both proxies and "devices", because they also transmit advertisements. This is why Shellys show up in the "configure devices" drop-down.

Local bluetooth adaptors (hci0, hci1 etc) and esphome proxies won't show up in that list, because they don't usually transmit advertisements.

Bermuda will use all scanners by default, so there's no gui for choosing which scanners to use. This will change a bit shortly when we fix #45 

So you don't need to specifically "enable" scanners, all you need to do is enable which "devices" you want Bermuda to set up sensors for. In the case of Shellys you can track them, but other scanners don't offer that (yet).

FYI, esphome just recently added the ability for sending advertisements while also acting as a proxy, so it will soon be able to do that too, but I haven't had a chance to do any testing yet to see if it works well (it might be tricky to get the balance right, and the esp is already pretty busy with proxy work), but we'll see. That shouldn't affect you for what you want to get going right now, though.

Does that make sense?

**Comment by wappinghigh** [comment]
Sure that makes a lot of sense.
Ok how can I identify my 4th (hidden - ie non shelly) scanner? I have no idea what it is!
Second question are you developing a kind "of list" of compatible "devices"/device names? (Or can you point me to one?)
 - as mostly (other than ibeacon - which I assume are apple)  I have zero idea what they are...

**Comment by jleinenbach** [comment]
> Ok how can I identify my 4th (hidden - ie non shelly) scanner?

As a first step, you could use https://macvendors.com/ to look up the vendor of the chip.

**Comment by wappinghigh** [comment]
Dude this is pretty darn cool!!!!!! Thanks for all the help setting it up!

**Comment by agittins** [comment]
> How can I identify my 4th (hidden - ie non shelly) scanner?

On each device page, there's a show more button:
![image](https://github.com/user-attachments/assets/114db95c-6d7c-42bb-9c40-a946bf0ae45e)

Each of the "distance to" entities point to a single scanner, which should help you identify what scanners bermuda is seeing. I'll think about exposing this somewhere else, too - since it does make it easier to work out what's going on if it just tells you :-)

![image](https://github.com/user-attachments/assets/61e1654b-c341-407f-abfa-b0aab72de57f)

> a kind "of list" of compatible "devices"/device names? (Or can you point me to one?)

That's a really good question, partly because it's a difficult one to answer. So I've started a wiki page that tries to answer that :-) Thanks for the nudge!
https://github.com/agittins/bermuda/wiki/Supported-devices

Basically, the goal is to support tracking of **any** bluetooth device that sends advertisements, but this isn't always practical.


> Dude this is pretty darn cool!!!!!! Thanks for all the help setting it up!

Awesome! There's more coolness on the way, too :-)

**Comment by wappinghigh** [comment]
^ This is fantastic. It really is. The Shelly's are a game changer. They are working really well with Bermuda and being able to be identified straight away once BT was turned on was the clinger for me. Being hardwired and fixed (I have switch relays and i4 buttons throughout) they become fixed point 'beacons". Testing now - the iphone "distance to Shelly" works really snappy- this is so so cool. So much potential  here. Congrats on the project. Can't wait to roll out some automations based off this.

**Comment by agittins** [comment]
> The Shelly's are a game changer.

Yeah, I didn't even realise they did that - I was writing Bermuda just thinking about esphome - but then all these people who'd bought Shelly devices essentially just magically got bluetooth trackers for free, it's awesome! :-)

> Why do all the devices wander around different areas all the time 

The Shelly devices will probably wander because even though they are "in their own area" so to speak, they are being read against the **other** areas, because the proxy doesn't read its "own" signal. If the other proxies are a reasonable distance away, then even small changes in signal strength can make a big difference in "estimated" distance, so it's easy for say, the "hallway" shelly to flip between "lounge" and "bedroom" if both are around 15 m away, since it doesn't take much signal variation for it to appear closer to one or the other.

Ultimately though, the "area" sensor of a shelly is a bit of a nonsense, since it is really always in it's *own* area, not in the area of some other proxy.

You'll probably see other devices flip around a bit between areas, this is because rssi is a pretty unreliable way to measure distances, but there are things you can do to limit it - check out this wiki page for a bit more info https://github.com/agittins/bermuda/wiki/Troubleshooting#my-device-switches-between-areas-every-few-seconds

**Comment by wappinghigh** [comment]
^Ok that makes sense as my Shellys are 10-15M apart. 
All good 
I'll continue to test 
Next I am going to try locate a casio BT watch
Thanks for the time answering the Q and setting up the wikis
Awesome project
Onwards and upwards! 
Cheers

**Comment by wappinghigh** [comment]
Hi again. Is there a lot of writing of files with Bermuda? Will this fill up my data? Will it wear out my SSD? thanks again
